File: linked-list/LinkedList.kt
class Node(var data: Int) {
    var next: Node? = null
}

class LinkedList {
    private var head: Node? = null
    
    fun append(data: Int) {
        val newNode = Node(data)
        if (head == null) {
            head = newNode
            return
        }
        
        var current = head
        while (current?.next != null) {
            current = current.next
        }
        current?.next = newNode
    }
    
    fun prepend(data: Int) {
        val newNode = Node(data)
        newNode.next = head
        head = newNode
    }
    
    fun display() {
        var current = head
        while (current != null) {
            print("${current.data} -> ")
            current = current.next
        }
        println("null")
    }
    
    fun search(data: Int): Boolean {
        var current = head
        while (current != null) {
            if (current.data == data) return true
            current = current.next
        }
        return false
    }
}

fun main() {
    val list = LinkedList()
    list.append(1)
    list.append(2)
    list.prepend(0)
    list.display()
    println("Search 2: ${list.search(2)}")
}

File: linked-list/linked_list.go
package main
import "fmt"

type Node struct {
    data int
    next *Node
}

type LinkedList struct {
    head *Node
}

func (ll *LinkedList) Append(data int) {
    newNode := &Node{data: data}
    
    if ll.head == nil {
        ll.head = newNode
        return
    }
    
    current := ll.head
    for current.next != nil {
        current = current.next
    }
    current.next = newNode
}

func (ll *LinkedList) Prepend(data int) {
    newNode := &Node{data: data, next: ll.head}
    ll.head = newNode
}

func (ll *LinkedList) Display() {
    current := ll.head
    for current != nil {
        fmt.Print(current.data, " -> ")
        current = current.next
    }
    fmt.Println("nil")
}

func (ll *LinkedList) Search(data int) bool {
    current := ll.head
    for current != nil {
        if current.data == data {
            return true
        }
        current = current.next
    }
    return false
}

func main() {
    ll := &LinkedList{}
    ll.Append(1)
    ll.Append(2)
    ll.Prepend(0)
    ll.Display()
    fmt.Printf("Search 2: %t\n", ll.Search(2))
}